zerver/data_import/slack.py

```diff
@@ -823,10 +824,16 @@ def get_messages_iterator(
                     # skipping those messages is simpler.
                     continue
                 if message.get("mimetype") == "application/vnd.slack-docs":
-                    # This is a Slack "Post" which is HTML-formatted,
-                    # and we don't have a clean way to import at the
-                    # moment.  We skip them on import.
-                    continue
+                    # This is a Slack "Post" which is HTML-formatted
+                    file_url = message.get("url_private_download")
+                    response = requests.get(file_url)
+                    response.encoding = "utf-8"
+                    # response encoding to "utf-8" is required to specify encoding for response.text
+                    markdown_content = ["markdownify", "--heading-style", "atx"]
+                    # markdownify is used to convert text from HTML to Markdown format
+                    text = subprocess.check_output(markdown_content, input=response.text, text=True)
```

Comment:
Note that we used `subprocess` for `html2text` because it's GPL-licensed -- but `markdownify` is MIT licensed, and thus there's no reason to not import it and call it directly.

But I also think that we should not be rolling a whole separate html-to-markdown pipeline here.  If this particular HTML-to-markdown conversion is running into shortcomings of `html2text` (likely due to vaguaries of our Markdown impleentation), then so are the _other_ callsites of `html2text`.

We should probably:
1. Move `convert_html_to_markdown` from `zerver.lib.email_notifications` to someplace like `zerver.lib.markdown.from_html` or somesuch
2. Switch it to using `markdownify` (in-process via `import`, not `subprocess`)
3. Extend the conversion with whatever tweaks are necessary to cleanly import Canvas HTML
4. Add some tests for the behaviour

Reply:
> 1. Move `convert_html_to_markdown` from `zerver.lib.email_notifications` to someplace like `zerver.lib.markdown.from_html` or somesuch

Do we have to refactor these changes here or separately in the next PR?

Reply:
Each of those should be a separate commit in a single PR.

If you haven't already, please read our documentation on [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) and [writing reviewable PRs](https://zulip.readthedocs.io/en/latest/contributing/reviewable-prs.html).